apollo-federation/src/sources/connect/models.rs

```diff
@@ -135,30 +184,45 @@ impl Connector {
             .source
             .as_ref()
             .and_then(|name| source_arguments.iter().find(|s| s.name == *name));
-
         let source_name = source.map(|s| s.name.clone());
+
+        // Create our transport
         let connect_http = connect
             .http
             .as_ref()
             .ok_or_else(|| internal_error!("@connect(http:) missing"))?;
         let source_http = source.map(|s| &s.http);
-
         let transport = HttpJsonTransport::from_directive(connect_http, source_http)?;
-        let request_variables: HashSet<Namespace> = transport
-            .variable_references()
+
+        // Get our batch and error settings
+        let batch_settings = ConnectorBatchSettings::from_directive(&connect);
+        let connect_errors = connect.errors.as_ref();
+        let source_errors = source.and_then(|s| s.errors.as_ref());
+        let error_settings = ConnectorErrorsSettings::from_directive(connect_errors, source_errors);
+
+        // Calculate which variables and headers are in use in the request
+        let request_references: HashSet<VariableReference<Namespace>> =
+            transport.variable_references().collect();
+        let request_variables: HashSet<Namespace> = request_references
+            .iter()
             .map(|var_ref| var_ref.namespace.namespace)
             .collect();
-        let request_headers = extract_header_references(transport.variable_references());
+        let request_headers = extract_header_references(request_references);
 
-        let response_variables: HashSet<Namespace> = connect
+        // Calculate which variables and headers are in use in the response (including errors.message and errors.extensions)
+        let response_references: HashSet<VariableReference<Namespace>> = connect
             .selection
             .variable_references()
+            .chain(error_settings.variable_references())
+            .collect();
```

Comment:
You don;t seem to need this collect as you are iterating over it in the next line again